### DISCUSSION TITLE
root login disabled

### DISCUSSION THREAD

**Discussion by grahamperrin** [helloSystem/hello#378] [General]
@probonopd please: can you remember where, long ago, we had the discussion about the root user? 

(Your decision for helloSystem to be Apple-like.) 

I thought it was a GitHub issue, but I can't find it. 

Thank you

<https://github.com/helloSystem/ISO/blob/f4f8c43f0888c2c6e0de5621a1c0fce132411082/overlays/uzip/furybsd-live-settings/files/usr/local/bin/furybsd-install#L120-L126>

```text
# Disable direct root login; users in the wheel group can use sudo instead
echo "Disabling direct root login; users in the wheel group can use sudo instead"
# "Star out" the root password; unlike locking the password this allows sudo su
chroot "${FSMNT}" env EDITOR=ed vipw <<EOF
/^root/s/root:[^:]*:/root:*:/
wq
EOF
```

Cross-reference: <https://github.com/helloSystem/ISO/commit/a59de6af82bb52f6d91053896e052f31fce44901>

>  Ensure users are never dropped into vi

**Comment by probonopd** [comment]
For the installed system, I think we don't want a password for root, but we also don't want it to be possible to log in as root, only to become root using `su`. So I think this is how it is now, and as such it works as intended? Or am I wrong?

> Ensure users are never dropped into vi

This may not have been achieved entirely for the root users and in all circumstances yet. So any suggestions for improving this for root and non-root users are welcome.

**Reply by grahamperrin** [reply]
Configuring helloSystem for [freebsd-update(8)](https://www.freebsd.org/cgi/man.cgi?query=freebsd-update&sektion=8&manpath=FreeBSD) to _not_ use [vi(1)](https://www.freebsd.org/cgi/man.cgi?query=vi&sektion=1&manpath=FreeBSD) is **a struggle, with root login disabled**. 

Whilst [freebsd-update(8): prevent vi, prefer ee : freebsd](https://is.reddit.com/r/freebsd/comments/xbycd8/freebsdupdate8_prevent_vi_prefer_ee/) is notionally _answered_: 

* the struggle to identify relevant files, and the best variable for use within those files, continues, in the context of [sudo(8)](https://www.freebsd.org/cgi/man.cgi?query=sudo&sektion=8&manpath=FreeBSD-Ports).

**Reply by grahamperrin** [reply]
> become root using `su`. 

Sorry, I can't remember how this is done. 

### Postcript

Found: <https://old.reddit.com/r/freebsd/comments/xbycd8/-/io6qf78/?context=2>

**Reply by probonopd** [reply]
>> become root using su.

> Sorry, I can't remember how this is done.

Just like on Ubuntu or on the Mac: `sudo su`
